bitnami/apisix/templates/tls-secret.yaml

```
@@ -0,0 +1,142 @@
+  {{ .Values.dataPlane.tls.certCAFilename }}: {{ include "common.secrets.lookup" (dict "secret" (printf "%s-tls" (include "apisix.data-plane.fullname" .)) "key" .Values.dataPlane.tls.certCAKeyFilename "defaultValue" $ca.Cert "context" $) }}
+  {{- else }}
+  {{- if .Values.dataPlane.tls.cert }}
+  {{ .Values.dataPlane.tls.certFilename }}: {{ .Values.dataPlane.tls.cert | b64enc }}
+  {{- end }}
+  {{- if .Values.dataPlane.tls.key }}
+  {{ .Values.dataPlane.tls.certKeyFilename }}: {{ .Values.dataPlane.tls.key | b64enc }}
+  {{- end }}
+  {{- if .Values.dataPlane.tls.ca }}
+  {{ .Values.dataPlane.tls.certCAFilename }}: {{ .Values.dataPlane.tls.ca | b64enc }}
+  {{- end }}
+  {{- end }}
+---
+{{- end }}
+{{- if and .Values.ingressController.enabled .Values.ingressController.tls.enabled (not .Values.ingressController.tls.existingSecret) }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ printf "%s-tls" (include "apisix.ingress-controller.fullname" .) }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/part-of: apisix
+    app.kubernetes.io/component: ingress-controller
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+type: kubernetes.io/tls
+data:
+  {{- if .Values.ingressController.tls.autoGenerated }}
+  {{- $cert := genSignedCert (include "apisix.ingress-controller.fullname" .) nil (list (include "apisix.ingress-controller.fullname" .) (printf "%s.%s" (include "apisix.ingress-controller.fullname" .) (include "common.names.namespace" .)) (printf "%s.%s.svc" (include "apisix.ingress-controller.fullname" .) (include "common.names.namespace" .)) (printf "%s.%s.svc.%s" (include "apisix.ingress-controller.fullname" .) (include "common.names.namespace" .) .Values.clusterDomain)) 365 $ca }}
+  {{ .Values.ingressController.tls.certFilename }}: {{ include "common.secrets.lookup" (dict "secret" (printf "%s-tls" (include "apisix.ingress-controller.fullname" .)) "key" .Values.ingressController.tls.certFilename "defaultValue" $cert.Cert "context" $) }}
+  {{ .Values.ingressController.tls.certKeyFilename }}: {{ include "common.secrets.lookup" (dict "secret" (printf "%s-tls" (include "apisix.ingress-controller.fullname" .)) "key" .Values.ingressController.tls.certKeyFilename "defaultValue" $cert.Key "context" $) }}
+  {{ .Values.ingressController.tls.certCAFilename }}: {{ include "common.secrets.lookup" (dict "secret" (printf "%s-tls" (include "apisix.ingress-controller.fullname" .)) "key" .Values.ingressController.tls.certCAKeyFilename "defaultValue" $ca.Cert "context" $) }}
```

The lookup key is wrong:
```suggestion
  {{ .Values.ingressController.tls.certCAFilename }}: {{ include "common.secrets.lookup" (dict "secret" (printf "%s-tls" (include "apisix.ingress-controller.fullname" .)) "key" .Values.ingressController.tls.certCAFilename "defaultValue" $ca.Cert "context" $) }}
```